### PR TITLE
Use same Modal component

### PR DESCRIPTION
**Pull request by ladytrekker**
## Description

This is another preparation PR for ticket #9 
This PR adds a new generic component for App "Modal" and refactored all occasions where a modal is shown to use this same component.

Please Note: I ignored the ClaimExceededModal as this is only relevant for PKlima view which is not maintained no more.

## Related Ticket

Part of #9 

## Changes

| Before  | After  |
|---------|--------|
| Bildschirmfoto 2022-07-19 um 12 35 08|Bildschirmfoto 2022-07-19 um 12 44 15 |
|Bildschirmfoto 2022-07-19 um 12 49 11 | Bildschirmfoto 2022-07-19 um 13 00 39 | 
| |Bildschirmfoto 2022-07-19 um 13 49 12  | 
| Bildschirmfoto 2022-07-19 um 13 05 18 |
Bildschirmfoto 2022-07-19 um 13 25 58<br>
Bildschirmfoto 2022-07-19 um 13 26 09<br>
Bildschirmfoto 2022-07-19 um 13 26 21<br>
Bildschirmfoto 2022-07-19 um 13 26 32 | 

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: app/components/Modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+interface Props {
+  title: React.ReactNode;
+  onToggleModal?: () => void;
+  className?: string;
+}
+
+export const Modal: FC<Props> = (props) => {
+  const showCloseButton = !!props.onToggleModal;
+  console.log(showCloseButton);
```
Comment (Atmosfearful) [inline app/components/Modal/index.tsx:16]: 👽

File: app/components/Modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  return (
+    <>
+      <div
+        className={cx(styles.modalBackground, {
+          showCloseButton,
+        })}
+        onClick={props.onToggleModal}
+      />
+      <div className={styles.modalContainer}>
+        <div className={cx(styles.modalContent, props.className)}>
```
Comment (Atmosfearful) [inline app/components/Modal/index.tsx:26]: always a dilemma which `div` to put the `props.className` on.

I've seen components get gross
```
props.titleClassName
props.containerClassName
```

but I really can't think of a better way

**Review by Atmosfearful — APPROVED** [review]
rad 💯

File: app/components/Modal/index.tsx (hunk 1 above)
Comment (ladytrekker) [reply app/components/Modal/index.tsx:16]: done

**Pull request merged**